src/features/account/MoreOptionMenu.test.tsx

```diff
@@ -109,10 +109,34 @@ describe('MoreOptionMenu', () => {
     navigator.clipboard.writeText.mockResolvedValue(undefined);
     const { getByTestId } = await renderMenuWithStore();
 
-    act(() => {
-      getByTestId('copy-profile-link').click();
+    await act(async () =>
```

Comment:
This need the double `act` to make state completely updated in toast and store. otherwise, error will throw for toastProvider.

Reply:
good to know that

Reply:
btw why not mock the toastProvider? or just mock the hook, with mocked provider , you may not need double act / wait

Reply:
That is because i need to assert the contents in toast, if moke ToastProvider (also dont know what internal functions or state inside).  so i rather than use double act() to make test more robust instead of mocking ToastProvider.

Reply:
you can use my reference
const { toastSpy } = buildToastSpy();
expect(toastSpy).toHaveBeenCalledWith(
        expect.objectContaining({
          title: 'Failed to sign the message',
          description: 'sign error',
        }),
      );

Reply:
As what we discuss,  something happen in
```
navigator.clipboard
      .writeText(`${window.location.origin}/account/?address=${subjectAddress}`)
      .then(() => {
        toast({
          title: t`Copied`,
          description: t`Profile link copied to clipboard`,
          status: 'success',
          duration: 2000,
          isClosable: true,
        });
      })
      ```
      which is promise to make toast state update late.